test/mailbox/model/mailbox_node_test.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:jmap_dart_client/jmap/core/id.dart';
+import 'package:jmap_dart_client/jmap/mail/mailbox/mailbox.dart';
+import 'package:model/mailbox/presentation_mailbox.dart';
+import 'package:tmail_ui_user/features/mailbox/presentation/model/mailbox_node.dart';
+
+void main() {
+  group('mailbox tree test', () {
```

Comment:
missing case:
- what happen to add child with the mailbox have `parentId` is not item's `id`
- what happen to add child with the mailbox have `parentId` is not null.
- what happen to add to the node with the existed child
- what happen to add the duplicated item to the node with the existed child
- what happen to add child to the node without children